infra/auth-server/src/utils/verify-sign.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const Web3Utils = require('web3-utils')
+
+const Eth = require('web3-eth')
+const Web3Eth = new Eth()
+
+const stringify = require('json-stable-stringify')
+
+const logger = require('../logger')
+
+/**
+ * Verifies the sign of message
+ *
+ * @param {String} address ETH address of the signer
+ * @param {String} signature The actual signature
+ * @param {Object} payload The data that was signed
+ * @returns {Boolean} `true` if `signature` is signed by `address`
+ *                    on `payload` in the last 15 minutes;
+ *                    `false` otherwise
+ */
+const verifySign = ({ address, signature, payload }) => {
+  try {
+    const now = Date.now()
+    const signExpLimit =
+      (parseInt(process.env.SIGN_EXPIRES_IN) || 15) * 1000 * 60
```

Comment:
Yay! I like this time much better. We'll just have to see how accurately people's clocks are set in the wild.

We should be okay on mobile though, since those are set automaticly.